### ISSUE TITLE
Consider splitting configuration of foreign types into a new `generate!` macro config key

### ISSUE THREAD

**Issue by rylev** [bytecodealliance/wit-bindgen#992]
In https://github.com/bytecodealliance/wit-bindgen/pull/972 the default for whether bindings would be generated for foreign (i.e., not in the same package as the target world) types was flipped from being generated by default to not. Users could still opt-into generating code for foreign types by using the `generate` keyword.

The use of a special keyword, however, might not be the most discoverable. Instead we may wish to introduce a new key in the `generate!` macro. 

In short the following...

```rust
with: {
    "wasi:io/poll": wasi::io::poll,
    "some:package/my-interface": generate,
},
```

...would become the following...

```rust
with: {
    "wasi:io/poll": wasi::io::poll,
},
generate: [
    "some:package/my-interface"
]
```

**Comment by alexcrichton** [comment]
Seems reasonable to me!
